lib/astarte_device_sdk/device.c

```diff
@@ -103,6 +103,16 @@ astarte_result_t astarte_device_new(astarte_device_config_t *cfg, astarte_device
     handle->cbk_user_data = cfg->cbk_user_data;
 
     // Initializing the connection hashmap and status flags
+
+    bool synchronization_completed = false;
+#if defined(CONFIG_ASTARTE_DEVICE_SDK_PERMANENT_STORAGE)
+    ares = astarte_device_caching_synchronization_get(&synchronization_completed);
+    if ((ares != ASTARTE_RESULT_OK) && (ares != ASTARTE_RESULT_NOT_FOUND)) {
+        ASTARTE_LOG_ERR("Synchronization state getter failure %s.", astarte_result_to_name(ares));
+        goto failure;
+    }
+#endif
+    handle->synchronization_completed = synchronization_completed;
```

Comment:
Use `synchronization_completed` directly from handle, without allocate different variable.
```suggestion
    handle->synchronization_completed = false;
#if defined(CONFIG_ASTARTE_DEVICE_SDK_PERMANENT_STORAGE)
    ares = astarte_device_caching_synchronization_get(&handle->synchronization_completed);
    if ((ares != ASTARTE_RESULT_OK) && (ares != ASTARTE_RESULT_NOT_FOUND)) {
        ASTARTE_LOG_ERR("Synchronization state getter failure %s.", astarte_result_to_name(ares));
        goto failure;
    }
#endif
```